comp/core/status/statusimpl/status.go

```diff
@@ -0,0 +1,315 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package statusimpl implements the status component interface
+package statusimpl
+
+import (
+	"bytes"
+	"embed"
+	"encoding/json"
+	"io"
+	"path"
+	"sort"
+	"text/template"
+
+	"go.uber.org/fx"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+//go:embed templates
+var templatesFS embed.FS
+
+type dependencies struct {
+	fx.In
+	Config config.Component
+	Params status.Params
+
+	Providers       []status.Provider       `group:"status"`
+	HeaderProviders []status.HeaderProvider `group:"header_status"`
+}
+
+type statusImplementation struct {
+	sortedHeaderSection []status.HeaderProvider
+	sortedSections      []string
+	sortedProviders     map[string][]status.Provider
+	providers           []status.Provider
+	headerProvider      []status.HeaderProvider
+}
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newStatus),
+	)
+}
+
+func sortByName(providers []status.Provider) []status.Provider {
+	sort.SliceStable(providers, func(i, j int) bool {
+		return providers[i].Name() < providers[j].Name()
+	})
+
+	return providers
+}
+
+func newStatus(deps dependencies) (status.Component, error) {
+	sections := []string{}
+
+	for _, provider := range deps.Providers {
+		if !present(provider.Section(), sections) {
+			sections = append(sections, provider.Section())
+		}
+	}
+
+	sortedProviders := map[string][]status.Provider{}
+
+	for _, provider := range deps.Providers {
+		providers := sortedProviders[provider.Section()]
+		sortedProviders[provider.Section()] = append(providers, provider)
+	}
+
+	for section, providers := range sortedProviders {
+		sortedProviders[section] = sortByName(providers)
+	}
+
+	sortedHeaderSection := deps.HeaderProviders
+	sort.SliceStable(sortedHeaderSection, func(i, j int) bool {
+		return sortedHeaderSection[i].Index() < sortedHeaderSection[j].Index()
```

Comment:
Should we sort by `Index()` and the `Name()` ?

Reply:
The header provider does not have a name property. We sort by index to make sure they are displayed in the correct order. So I'm not sure we should order by `Name`

Now I realize that we should add the `Name` to the header provider so we can add the name section to the output